src/currencyRatesFileGen.js

```diff
@@ -262,21 +258,49 @@ function createDocumentParams(bucket, filename, documentObj, expires) {
     }
 }
 
-async function fetchMultipleUrlsAsPromise(fetchUrls){
-    const responses = [];
-    for (let url of fetchUrls) {
-        const promise = new Promise((resolve, reject) => {
-            requestJSONData(url, resolve, reject);
-        });
-        const data = await promise;
-        validateCurrencyData(data);
-        responses.push(data);
+async function fetchCurrencyResponsesAsPromise() {
+    const promise = new Promise((resolve, reject) => {
+        requestXMLData('https://www.ecb.europa.eu/stats/eurofxref/eurofxref-daily.xml', resolve, reject);
+    });
+    const data = await promise;
+    const currencyData = data["gesmes:Envelope"]["Cube"][0]["Cube"][0];
+    let currencyObjects = normalizeData(currencyData)
+    currencyObjects.forEach(currency => validateCurrencyData(currency))
+
+    return currencyObjects;
+}
+
+function normalizeData(currencyData) {
+    if (!currencyData) {
+        throw new Error(`Error: currency data is not present`);
     }
-    return responses;
+    const currencyRates = currencyData["Cube"];
+    const currencyDate = currencyData["$"]["time"];
+    const currencyObjects = currencyRates.map(data => {
+        return Object.create({
+            currency: data["$"]["currency"],
+            rate: data["$"]["rate"]
+        });
+    })
+    let requestedCurrencies = currencyObjects.filter(curObj => fromCurrencies.includes(curObj.currency));
+    return requestedCurrencies.map(obj => createCurrencyResponse(obj, currencyObjects, currencyDate));
+}
+
+function createCurrencyResponse(currencyObject, currencyData, currencyDate) {
+    let response = Object.create({});
+    response["base"] = currencyObject.currency;
+    response["date"] = currencyDate;
+    let rate = Object.create({});
+    currencyData.forEach(data => {
+        rate[data.currency] = data.rate / currencyObject.rate;
+    })
+    rate["EUR"] = 1/ currencyObject.rate;
```

Comment:
Pls add comment here why we need to add separate currency. Or even more, extract missing currencies to array and process them here. Up to you.